rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,498 @@
+            }
+
+        class HVACRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section19Rule21.RMDRule.HVACRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    required_fields={
+                        "$": ["fan_system"],
+                        "fan_system": [
+                            "supply_fans",
+                            "minimum_outdoor_airflow",
+                            "maximum_outdoor_airflow",
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                hvac_b = context.baseline
+                hvac_id_b = hvac_b["id"]
+
+                climate_zone = data["climate_zone"]
+                ER_not_req_for_heating_sys = climate_zone in APPLICABLE_CZ_ZONES
+
+                zone_data = data["zone_data"][hvac_id_b]
+                serves_zones_with_systems_likely_exhausting_toxic_etc = zone_data[
+                    "serves_zones_with_systems_likely_exhausting_toxic_etc"
+                ]
+                serves_kitchen_space = zone_data["serves_kitchen_space"]
+                all_lighting_space_types_defined = zone_data[
+                    "all_lighting_space_types_defined"
+                ]
+                all_ventilation_space_types_defined = zone_data[
+                    "all_ventilation_space_types_defined"
+                ]
+                ER_modeled_p = data["ER_modeled_p"]
+                serves_zones_that_have_dehumid_heat_recovery = data[
+                    "serves_zones_that_have_dehumid_heat_recovery"
+                ]
+                sys_type_heating_only = data["sys_type_heating_only"]
+                hvac_systems_and_assoc_zones_largest_exhaust_source = data[
+                    "hvac_systems_and_assoc_zones_largest_exhaust_source"
+                ][hvac_id_b]
+                serves_zones_heated_to_60_or_higher_p = data[
+                    "serves_zones_heated_to_60_or_higher_p"
+                ]
+
+                fan_sys_b = hvac_b["fan_system"]
+                supply_airflow_b = sum(
+                    [
+                        supply_fan_b.get("design_airflow", ZERO.FLOW)
+                        for supply_fan_b in fan_sys_b["supply_fans"]
+                    ]
+                )
+
+                min_outdoor_airflow_b = fan_sys_b["minimum_outdoor_airflow"]
+                max_outdoor_airflow_b = fan_sys_b["maximum_outdoor_airflow"]
+
+                OA_fraction_b = (
+                    min_outdoor_airflow_b / supply_airflow_b
+                    if supply_airflow_b != ZERO.FLOW
+                    else 0.0
+                )
+
+                ER_modeled_b = fan_sys_b.get("air_energy_recovery") not in (
+                    None,
+                    ENERGY_RECOVERY.NONE,
+                )
```

Comment:
We can be more consistent with the method here and the above one that calculates for `ER_modeled_p`